modules/core/src/main/scala/fs2/kafka/internal/WithProducer.scala

```diff
@@ -19,70 +19,54 @@ private[kafka] object WithProducer {
   def apply[F[_], K, V](
     settings: ProducerSettings[F, K, V]
   )(
-    implicit F: Sync[F],
-    context: ContextShift[F]
+    implicit F: Sync[F]
   ): Resource[F, WithProducer[F]] = {
-    val blockerResource =
-      settings.blocker
-        .map(Resource.pure[F, Blocker])
-        .getOrElse(Blockers.producer)
+    Resource[F, WithProducer[F]] {
+      settings.createProducer.map { producer =>
+        val withProducer =
+          new WithProducer[F] {
+            override def apply[A](f: KafkaByteProducer => F[A]): F[A] =
+              F.defer {
+                f(producer)
```

Comment:
same problem with `blocker`

Reply:
Although maybe you are wrapping that somewhere else?

Reply:
I've changed from `delay` to `blocking` at the call site. (What I suggested above won't straightforwardly work where this is used by `TransactionalProducer`.